### ISSUE TITLE
Is MinVIS truly online?

### ISSUE THREAD

**Issue by timmeinhardt** [NVlabs/MinVIS#2]
First of all, congratulations on this paper. It was a very interesting read. However, I think technically MinVIS can not be considered an online method. You are processing each frame separately but an online method must not include information of future frames for the decision making on the current frame. In this line

https://github.com/NVlabs/MinVIS/blob/3038871b86c4f6b6bd8bfe59805fb8d43f868bc1/minvis/video_maskformer_model.py#L308

you compute mean scores for each query and class across the entire sequence. These scores are later used for the topk selection of final outputs. This processing is not online but offline as you utilize information of all frames at once. Please clarify what I might be misunderstanding or your point of view on the matter. Thank you!

**Comment by JialianW** [comment]
To my understanding, this is only for evaluation purpose. VIS requires each tacklet to have a score in order to compute mAP. While the most straightforward way to get a score for a tracklet is to average across frames. I don't think this would be necessary when using in real streaming applications.

**Comment by timmeinhardt** [comment]
I agree that the GT and prediction file design of YouTube-VIS and OVIS both invite to process their data in an offline fashion but it is possible to generate appropriate full-sequence tracks even for true online methods. For example, IDOL not only processes sequences online but never uses information of future frames for the mask/score prediction of the current frame. This requires full frame to frame track management and to suffice the VIS GT format they retroactively fill in missing/occluded frames with zeros (see [here](https://github.com/wjf5203/VNext/blob/51c3e261b55407f78fcdc81aa73ca4493de2ba32/projects/IDOL/idol/idol.py#L436)).

In a real world streaming application this is not necessary but when it comes to evaluation and comparability of methods this aspect is important and computing scores over the full sequence is usually considered to be offline. Using global information/averaging would probably benefit IDOL as well. Hence, I think the comparison is not fair.
